staging/src/k8s.io/apiserver/pkg/endpoints/handlers/watch.go

```diff
@@ -114,6 +115,11 @@ func serveWatch(watcher watch.Interface, scope *RequestScope, mediaTypeOptions n
 		defer runtime.AllocatorPool.Put(memoryAllocator)
 		embeddedEncoder = runtime.NewEncoderWithAllocator(encoderWithAllocator, memoryAllocator)
 	}
+	var tableOptions *metav1.TableOptions
+	if passedOptions, ok := options.(*metav1.TableOptions); ok {
```

Comment:
if all code paths that propagate options here give us TableOptions, should this treat a non-nil non-TableOptions as an error? if we ever add another options type, I want to make sure we don't forget this piece